Kconfig.zephyr

```diff
@@ -717,6 +717,25 @@ config BUILD_OUTPUT_STRIP_PATHS
 	  Debuggers usually have a path mapping feature to ensure the files are
 	  still found.
 
+config CHECK_INIT_PRIORITIES
+	bool "Build time initialization priorities check"
+	help
+	  Check the build for initialization priority issues by comparing the
+	  initialization priority in the build with the device dependency
+	  derived from the devicetree definition.
```

Comment:
At some point, we need to make this `default y`.

Reply:
Yeah, next step for this is to do a  `default y` CI run and start engaging with the maintainers of whatever fails.